[GeoMechanicsApplication] Add a linear elastic constitutive law for line interfaces (#12654)
KratosMultiphysics/Kratos: applications/GeoMechanicsApplication/custom_constitutive/README.md

# Constitutive laws

## Incremental linear elastic interface law

> The constitutive law for an interface element relates tractions $\tau$ to relative displacement $\Delta u$.

Reviewer: I would suggest to add the word 'linearly' to the sentence below to be more explicit about the nature of the relation:
```suggestion
The constitutive law for an interface element linearly relates tractions $\tau$ to relative displacement $\Delta u$.
```

Author: Yes, being more explicit about the nature of the relation may be clearer. Done.

Relative displacement for interface element is the differential motion between the two sides of the interface. As a

> consequence the relative displacement has unit of length [L] and the stiffness has unit of force over cubic length [F/L^3].

Reviewer 2: Preferly to be "m" and "N" replacing of L and F.
I know that you don't want to relate it to any specific unit system like SI, and wanted to write it as symbolic. But in the standard units symbols, F is a dependent unit and it hasn't specific symbol. The force then becomes [M L T^-2]. 

Moreover: The unit is preferly to be written as  "$\mathrm{[UNIT]}$" , to account for powers, For example,  "$\mathrm{[F/L^3]}$" or "$\mathrm{[N/m^3]}$"

Author: We will make use of the formula editor for correct formatting. However, in engineering it is very common to use the derived unit for Force i.s.o. mass * length * time^-2. To be independent of the system used ( S.I or imperical etc. ) we will express in L, F, t, T (i.e. quantities) i.s.o. mm, kN, days and Fahrenheit (i.e. some units).

> consequence the relative displacement has unit of length [$\mathrm{L}$] and the stiffness has unit of force over cubic length [$\mathrm{F/L^3}$].

Reviewer 3: These mathrms don't render correctly, I think reversing dollar and `[` should work `$[\mathrm{F/L^3}]$` (would look like $[\mathrm{F/L^3}]$), adding a space between `[` and `$` also works

Author: Good catch. Done.

> consequence the relative displacement has unit of length [$\mathrm{L}$] and the stiffness has unit of force over cubic length [$\mathrm{F/L^3}$].

Reviewer: ```suggestion
consequence the relative displacement has unit of length $[\mathrm{L}]$ and the stiffness has unit of force over cubic length $[\mathrm{F/L^3}]$.
```

Author: Done.

### Relative displacement and traction

> In 2D plane strain y is the opening/closing direction of the interface, while differential motion in the tangential direction

Reviewer 2: "y" is a variable, so it has to be in italic (equation form). So, i think better to use "$y$" instead.

Author: Done.

> gives shear. Like for a continuum, normal behaviour is placed first in the relative displacement and traction vector. Shear 

Reviewer: ```suggestion
gives shear. Similar to the continuum, the normal behaviour is placed first in the relative displacement and traction vector. The shear 
```

Author: Done.

> is placed after the normal motion or traction.

Reviewer: ```suggestion
behavior follows after the normal motion or traction in these vectors.
```

Author: Done.

$$ \Delta u = \begin{bmatrix} \Delta u_y \\ \Delta u_x \end{bmatrix} $$

> $$ \tau = \begin{bmatrix} \tau_{yy} \\ \tau_{xy} \end{bmatrix} $$

Reviewer 2: I am not sure whether I am correct in the case of interface, but in the normal stress tensors, the normal components are defined by $\sigma_{xx}$. $\sigma_{yy}$ etc. and shears by $\tau_{xy}$. Is it also here applicable?

Author: The name for stress in an interface is traction, and it is denoted by $\tau$. The symbol $\sigma$ is not used in this context at all.

> $$ \tau = \begin{bmatrix} \tau_{yy} \\ \tau_{xy} \end{bmatrix} $$

Reviewer: This is more style and I only suggest it because sometimes you just want to quickly check the formula and what the parameters stand for. So I would suggest adding the following: (if correct to my understanding)

> Where:

- $\Delta u_𝑦$: Relative displacement in the 𝑦-direction (normal to the interface).
- $\Delta u_𝑥$: Relative displacement in the 𝑥-direction (tangential to the interface).
- $\tau_{𝑦𝑦}$: Traction in the 𝑦-direction (normal to the interface).
- $\tau_{𝑥𝑦}$: Shear traction in the 𝑥-direction (tangential to the interface).

Author: Done.

### 2D Elastic constitutive tensor

> 

Reviewer: I would suggest to add something here just for a clearer documentation:

> The elastic behavior of the interface is characterized by the elastic constitutive tensor (or 2D Elastic constitutive tensor) 𝐶, which relates the traction and relative displacement vectors. The constitutive tensor in 2D is expressed as:

Author: Done.

$$ C = \begin{bmatrix} C_{yy} & 0     \\

>                        0     & C_{xy} \end{bmatrix}$$

Reviewer 2: What are the $C_{xx}$ and $C_{xy}$? I know they are stiffness parameters which are mentioned in the keywords, but preferly to be explicitely mentioned what they are, and what are their units.

Author: Added the dimension of the stiffness values to be more explicit about what is expected. The normal and shear stiffness terms are clear from the input parameter names.

> Where $C_{yy}$ is input as `INTERFACE_NORMAL_STIFFNESS` and $C_{xy}$ is input as `INTERFACE_SHEAR_STIFFNESS`. Both stiffness values have dimension $\mathrm{F/L^3}$.

Reviewer: Also for this one:

> Where:

- $C$: Represents the 2D elastic constitutive tensor.
- $C_{𝑦𝑦}$: Represents the `INTERFACE_NORMAL_STIFFNESS`, which characterizes the stiffness in the normal direction.
- $C_{𝑥𝑦}$: Represents the `INTERFACE_SHEAR_STIFFNESS`, which characterizes the stiffness in the tangential direction.

 Both stiffness values have dimension $\mathrm{F/L^3}$.

Author: Done.

### Incremental relation

> 

Reviewer: Also here, I suggest again to add something like:

> The relation between the traction and the relative displacement at a given time increment is given by the following incremental equation:

Author: Done.

$$ \tau_{t + \Delta t} = \tau_t + C \cdot \Delta u $$

> 

Reviewer: Also for this one:

> Where:

- $\tau_{t + \Delta t}$: The traction at the updated time 𝑡 + Δ𝑡.
- $\tau_t$: The traction at the current time 𝑡.
- $C$: The elastic constitutive tensor.
- $\Delta u$: The incremental relative displacement vector.

Author: Done.
